### PR TITLE
Slightly more padding in dense tables

### PR DESCRIPTION
**Pull request by stefanalund**
## Hey, I just made a Pull Request!

I felt like the new `dense` mode in the catalog is a bit toooooo dense. @freben you with me? 😛 

Old:
Screenshot 2020-12-04 at 15 43 09

New:
Screenshot 2020-12-04 at 15 42 50

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by freben — APPROVED** [review]
Fractions?! What is this madness?

@andrewthauer Better, right? :)

**Comment by Fox32** [comment]
Don't forget the changeset 😉

**Comment by stefanalund** [comment]
Still a Draft PR @Fox32 😎I was expecting this to be shot down 😅

File: .changeset/small-comics-cover.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/theme': minor
```
Comment (Fox32) [inline .changeset/small-comics-cover.md:2]: Should just be `patch` while we are below a `1.0.0` release

**Review by Fox32 — APPROVED** [review]
👍 Looks good

**Pull request merged**